contracts/Lockbox.cdc

```diff
@@ -0,0 +1,468 @@
+/*
+
+    Lockbox implements the functionality required to manage FLOW
+    buyers locked tokens from the token sale.
+
+    Each token holer gets two accounts. One account is their locked token
+    account. It will be jointly controlled by the user and Dapper Labs.
+    No actions will be able to be performed with this account without authorization
+    from Dapper Labs. Locked tokens are stored in this account and Dapper Labs
+    with authorize withdrawals every time a milestone is passed in the
+    vesting period.
+        
+    The second account is the unlocked user account. This account is
+    in full possesion and control of the user and they can do whatever
+    they want with it. This account will store a capability that allows
+    them to withdraw tokens when they become unlocked and also to
+    perform staking operations with their locked tokens.
+
+    When a user account is created, both accounts are initialized with
+    their respective objects, LockedTokenManager for the shared account,
+    and TokenHolder for the unlocked account. The user calls functions
+    on TokenHolder to withdraw tokens from the shared account and to 
+    perform staking actions with the locked tokens
+
+ */
+
+import FlowToken from 0x0ae53cb6e3f42a79
+import FungibleToken from 0xee82856bf20e2aa6
+import FlowIDTableStaking from 0xFLOWIDTABLESTAKINGADDRESS
+
+import StakingProxy from 0xSTAKINGPROXYADDRESS
+
+pub contract Lockbox {
+
+    pub event SharedAccountCreated(address: Address)
+    pub event UnlockedAccountCreated(address: Address)
+
+    /// path to store the locked token manager resource 
+    /// in the shared account
+    pub let LockedTokenManagerStoragePath: Path
+
+    /// Path to store the private capability for the token
+    /// manager
+    pub let LockedTokenManagerPrivatePath: Path
+
+    /// path to store the private locked token admin link
+    /// in the shared account
+    pub let LockedTokenAdminPrivatePath: Path
+
+    /// path to store the admin collection 
+    /// in the admin account
+    pub let LockedTokenAdminCollectionStoragePath: Path
+
+    /// path to store the token holder resource
+    /// in the unlocked account
+    pub let TokenHolderStoragePath: Path
+
+    /// Public path to store the capability that allows
+    /// reading the unlock limit of an account
+    pub let UnlockLimitPublicPath: Path
+
+    /// stored in Admin account to use to increase the unlock
+    /// token limit every time a vesting release happens
+    pub resource interface TokenAdmin {
+        pub fun increaseUnlockLimit(delta: UFix64)
+    }
+
+    /// stored in the shared account to manage access to the locked token vault
+    /// and to the staking/delegating resources
+    pub resource LockedTokenManager: FungibleToken.Receiver, FungibleToken.Provider, TokenAdmin {
```

Comment:
Do you think it would be more intuitive to flip the naming around here?

We could call the contract `LockedTokens` and then rename this resource to `Lockbox`. I feel like "lockbox" more accurately describes the actual resource that holds the tokens, and `LockedTokenManager` also feels a bit generic.

Reply:
yeah, I've been thinking something similar to this, but hadn't had the courage to do it, but I think I will now that you've mentioned it

Reply:
power to you!